src/pwmcmds.c

```diff
@@ -52,11 +96,26 @@ void
 command_schedule_pwm_out(uint32_t *args)
 {
     struct pwm_out_s *p = oid_lookup(args[0], command_config_pwm_out);
-    sched_del_timer(&p->timer);
-    p->timer.func = pwm_event;
-    p->timer.waketime = args[1];
-    p->value = args[2];
-    sched_add_timer(&p->timer);
+    struct pwm_event* new_event = (struct pwm_event*) mq_alloc_event();
+    new_event->waketime = args[1];
+    new_event->value = args[2];
+
+    if(!mq_event_insert(&p->queue, &new_event->event)) {
+        // queue was empty and a timer needs to be added
+
+        if(p->max_duration && p->current_value != p->default_value) {
+            // max_duration and currently not default value
+            // There has to be a safety timer
+            if(new_event->waketime > p->timer.waketime) {
```

Comment:
Doing direct comparisons of timers is not safe (due to integer rollover handling).  Timers should always be compared using `timer_is_before()`.